src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -1,10 +1,53 @@
-import React, { useCallback, useRef, useEffect } from 'react';
+import React, { Ref, useCallback, useRef, useEffect } from 'react';
 import { Vector3 } from 'three';
 import { ReactThreeFiber, PointerEvent, useThree } from 'react-three-fiber';
 import { clamp } from 'lodash-es';
+import { useMeasure } from 'react-use';
+import { useHeatmapState } from './store';
 
 const ZOOM_FACTOR = 0.95;
 
+export function useHeatmapSize<T>(
+  dims: [number, number],
+  axisOffset: [number, number]
+): [(elem: T) => void, { width: number; height: number } | undefined] {
+  const { hasOrthoCoords } = useHeatmapState();
+  const [wrapperRef, { width, height }] = useMeasure();
+
+  if (width === 0 && height === 0) {
+    return [wrapperRef, undefined];
+  }
+
+  if (!hasOrthoCoords) {
+    return [wrapperRef, { width, height }];
+  }
+
+  const [rows, cols] = dims;
+  const dimsRatio = rows / cols;
+
+  const [leftAxisWidth, bottomAxisHeight] = axisOffset;
+  const actualWidth = width - leftAxisWidth;
+  const actualHeight = height - bottomAxisHeight;
```

Comment:
`imageWidth` | `imageHeight` ? or `meshWidth` ?

Reply:
Oh I like `meshWidth`! -- I'm not a fan of the term `image`, as it's too much tied to JPEGs/PNGs/`<img>`... in my head.

Reply:
Ah actually, maybe not... I initially had `availableWidth` and `availableHeight`. I think I might revert to that, as it conveys the fact that I'm getting space available for the mesh, not the size of the mesh itself. Perhaps `targetWidth` and `targetHeight` should be the ones renamed to `meshWidth` and `meshHeight`?

Reply:
I wholly agree with `availableWidth` and renaming `target` to `mesh` :+1: